guide/qip/qip-simulator.rst

```diff
@@ -3,35 +3,53 @@
 
 .. _qip_simulator:
 
-****************
-Exact Simulation
-****************
+*********************************
+Operator-level circuit simulation
+*********************************
 
-Let's start off by defining a simple circuit which we use to demonstrate a few
-examples of circuit evolution. First, we read in the circuit from a **.qasm** file using
-the :func:`qutip.qip.qasm.read_qasm()` function.
-
-.. testcode::
+.. note::
 
-  from qutip.qip.qasm import read_qasm
-  from qutip import tensor, basis
+   Available from QuTiP 4.6
 
-  qc = read_qasm("w-state.qasm")
+Run a quantum circuit
+---------------------
 
-  qc.png
+Let's start off by defining a simple circuit which we use to demonstrate a few
+examples of circuit evolution. We take a circuit from
+`OpenQASM 2 <https://github.com/Qiskit/openqasm/blob/OpenQASM2.x/examples/W-state.qasm>`_
 
-.. image:: /gallery/qutip_examples/qip/images/w_state.png
+.. testcode::
 
-This circuit prepares the W-state :math:`\newcommand{\ket}[1]{\left|{#1}\right\rangle} \frac{\ket{001} + \ket{010} + \ket{100}}{\sqrt{3}}`.
+    from qutip.qip.circuit import QubitCircuit, Gate
+    from qutip.qip.operations import controlled_gate, hadamard_transform
+    def controlled_hadamard():
+        # Controlled Hadamard
+        return controlled_gate(
+            hadamard_transform(1), 2, control=0, target=1, control_value=1)
+    qc = QubitCircuit(N=3, num_cbits=3)
+    qc.user_gates = {"cH": controlled_hadamard}
+    qc.add_gate("QASMU", targets=[0], arg_value=[1.91063, 0, 0])
+    qc.add_gate("cH", targets=[0,1])
+    qc.add_gate("TOFFOLI", targets=[2], controls=[0, 1])
+    qc.add_gate("X", targets=[0])
+    qc.add_gate("X", targets=[1])
+    qc.add_gate("CNOT", targets=[1], controls=0)
+
+It corresponds to the following circuit:
+
+.. image:: /figures/qip/quantum_circuit_w_state.png
+
+We will add the measurement gates later. This circuit prepares the W-state :math:`\newcommand{\ket}[1]{\left|{#1}\right\rangle} \frac{\ket{001} + \ket{010} + \ket{100}}{\sqrt{3}}`.
```

Comment:
probably easier to read the state if you do an inline fraction (`(\ket{001} + \ket{010} + \ket{100})/3`) rather than this one